ratpack-test/src/main/java/ratpack/test/http/internal/DefaultMultipartForm.java

```diff
@@ -0,0 +1,396 @@
+
+    private final String _boundary;
+    private final Map<String, Set<String>> _fields;
+    private final Map<String, Set<File>> _files;
+
+    private Builder() {
+      this._boundary = generateBoundary();
+      this._fields = new TreeMap<>();
+      this._files = new TreeMap<>();
+    }
+
+    public String getBoundary() {
+      return _boundary;
+    }
+
+    public Map<String, Set<String>> getFields() {
+      return Collections.unmodifiableMap(_fields);
+    }
+
+    public Map<String, Set<File>> getFiles() {
+      return Collections.unmodifiableMap(_files);
+    }
+
+    public DefaultMultipartForm build() {
+      return new DefaultMultipartForm(new Data(_boundary, _fields, _files));
+    }
+
+    public Builder field(String name, String value) {
+      if (!_fields.containsKey(name)) {
+        _fields.put(name, new TreeSet<>());
+      }
+      _fields.get(name).add(value);
+
+      return this;
+    }
+
+    public Builder fields(Map<String, String> data) {
+      data.forEach(this::field);
+
+      return this;
+    }
+
+    public FileBuilder file() {
+      return new FileBuilder(this);
+    }
+
+    private Builder add(File file) {
+      String name = file.getField();
+      if (!_files.containsKey(name)) {
+        _files.put(name, new TreeSet<>());
+      }
+      _files.get(name).add(file);
+
+      return this;
+    }
+
+    private static String generateBoundary() {
+      return String.format("TEST%d", System.currentTimeMillis());
+    }
+
+  }
+
+  public final static class FileBuilder implements MultipartFileSpec {
+
+    private final Builder _builder;
+
+    private String _contentType;
+    private byte[] _data;
+    private String _encoding;
+    private String _field;
+    private String _name;
+
+    private FileBuilder(Builder builder) {
+      this._builder = builder;
+      this._contentType = "text/plain";
+      this._field = "file";
+      this._name = "filename.txt";
+    }
+
+    public Builder add() {
+      File file = new File(_field, _contentType, _encoding, _name, _data);
+      file.validate();
+
+      return _builder.add(file);
+    }
+
+    public FileBuilder contentType(String contentType) {
+      this._contentType = contentType;
+
+      return this;
+    }
+
+    public FileBuilder data(byte[] data) {
+      this._data = data;
+
+      return this;
+    }
+
+    public FileBuilder data(String data) {
+      try {
+        this._data = data.getBytes("UTF-8");
+      } catch(UnsupportedEncodingException uee) {
+        throw new IllegalArgumentException(uee);
+      }
+
+      return this;
+    }
+
+    public FileBuilder encoding(String encoding) {
+      this._encoding = encoding;
+
+      return this;
+    }
+
+    public FileBuilder field(String field) {
+      this._field = field;
+
+      return this;
+    }
+
+    public FileBuilder name(String name) {
+      this._name = name;
+
+      return this;
+    }
+
+  }
+
+  public final static class File implements Comparable<File> {
```

Comment:
Why are there underscores in the names of fields in this class? I don't think such naming convention is used in ratpack's codebase.